src/libs/actions/Policy.ts

```diff
@@ -630,6 +637,65 @@ function removeMembers(accountIDs: number[], policyID: string) {
     API.write(WRITE_COMMANDS.DELETE_MEMBERS_FROM_WORKSPACE, params, {optimisticData, successData, failureData});
 }
 
+function updateWorkspaceMembersRole(policyID: string, accountIDs: number[], newRole: typeof CONST.POLICY.ROLE.ADMIN | typeof CONST.POLICY.ROLE.USER) {
+    const previousPolicyMembers = {...allPolicyMembers};
+    const data: WorkspaceMembersRoleData[] = accountIDs
+        .map((accountID) => {
+            if (!allPersonalDetails?.[accountID]) {
+                return null;
+            }
+
+            return {
+                accountID,
+                email: allPersonalDetails?.[accountID]?.login,
+                role: newRole,
+            };
+        })
+        .filter((item): item is WorkspaceMembersRoleData => item !== null);
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBERS}${policyID}`,
+            value: {
+                ...data.reduce((acc: Record<number, {role: string}>, current) => {
```

Comment:
Why is this reduce necessary? Isn't the new role already updated in data?

Reply:
The `data` is an array of objects, but for Onyx key we need an object where keys are accountID's and with `{role:'...'}` object assigned - that's why we need to do this transformation using `reduce` method.